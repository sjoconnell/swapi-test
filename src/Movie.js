import React, { Component } from 'react';
import axios from 'axios';

class Movie extends Component {

  state = {
    characters: []
  }

  componentDidMount() {
    axios.all(
      this.props.film.characters
        .map(character => axios.get(character))
    ).then((responses) => {
      this.setState({
        characters: responses.map(response => response.data.name)
      })
    })

  }

  render() {

    return (
      <div>
        <h1>Single Movie!</h1>
      </div>
    )
  }
}

export default Movie