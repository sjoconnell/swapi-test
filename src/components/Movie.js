import React from 'react';

const getCharacters = (id) => {
  const characters = {
    1: ['Anakin Skywalker', 'Padme Amidala', 'Darth Maul'],
    2: ['Anakin Skywalker', 'Padme Amidala', 'Yoda'],
    3: ['Anakin Skywalker', 'Padme Amidala', 'Yoda'],
    4: ['Luke Skywalker', 'Leia Organa', 'Han Solo'],
    5: ['Luke Skywalker', 'Darth Vader', 'Yoda'],
    6: ['Luke Skywalker', 'Darth Vader', 'Leia Organa'],
    7: ['Rey', 'Finn', 'Kylo Ren']
  }
  return characters[id]
}

const Movie = ({film}) => {
  return (
    <div className="movie">
      <img className="movie-image" src={require(`../css/images/${film.title}.jpeg`)} alt={film.title}/>
      <div className="movie-info">
        <h1 className="movie-info-title">{film.title}</h1>
        <h2 className="movie-info-director">{film.director}</h2>
      </div>
      <div className="movie-characters">
        {
          getCharacters(film.episode_id)
            .map(character =>
              <div className="movie-characters-container" key={character}>
                <img className="movie-characters-image" src={require(`../css/images/${character}.png`)} alt={character}  />
                <div className="movie-characters-container-hover">{character}</div>
              </div>
            )
        }
      </div>
    </div>
  )
}

export default Movie