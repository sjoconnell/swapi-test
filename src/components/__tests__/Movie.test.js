import React from 'react';
import { shallow } from 'enzyme';
import Movie from '../Movie';

it('Movie renders correctly', () => {
  const component = shallow(<Movie />);
  expect(component).toMatchSnapshot();
});