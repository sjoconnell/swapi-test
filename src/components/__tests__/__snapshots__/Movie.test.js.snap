// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="movie"
>
    <img
        alt="BB8"
        className="movie-image"
        src="bb8.jpg"
    />
    <div
        className="movie-info"
    >
        <h1
            className="movie-info-title"
        />
        <h2
            className="movie-info-director"
        />
    </div>
    <div
        className="movie-characters"
    />
</div>,
  "nodes": Array [
    <div
      className="movie"
>
      <img
            alt="BB8"
            className="movie-image"
            src="bb8.jpg"
      />
      <div
            className="movie-info"
      >
            <h1
                  className="movie-info-title"
            />
            <h2
                  className="movie-info-director"
            />
      </div>
      <div
            className="movie-characters"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Movie />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="movie"
>
          <img
                    alt="BB8"
                    className="movie-image"
                    src="bb8.jpg"
          />
          <div
                    className="movie-info"
          >
                    <h1
                              className="movie-info-title"
                    />
                    <h2
                              className="movie-info-director"
                    />
          </div>
          <div
                    className="movie-characters"
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="movie"
>
          <img
                    alt="BB8"
                    className="movie-image"
                    src="bb8.jpg"
          />
          <div
                    className="movie-info"
          >
                    <h1
                              className="movie-info-title"
                    />
                    <h2
                              className="movie-info-director"
                    />
          </div>
          <div
                    className="movie-characters"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Movie />,
}
`;
